src/sentry/receivers/features.py

```diff
@@ -122,7 +122,7 @@ def record_event_processed(project, event, **kwargs):
     # Check to make sure more the ip address is being sent.
     # testing for this in test_no_user_tracking_for_ip_address_only
     # list(d.keys()) pattern is to make this python3 safe
-    if user_context and list(user_context.keys()) != ["ip_address"]:
+    if user_context and list(user_context.keys()) != ["ip_address", "sentry_user"]:
```

Comment:
Fixes the test `test_no_user_tracking_for_ip_address_only`.

Relay now extracts `sentry_user` automatically from the IP address, and both should be filtered out together and not trigger user tracking.